File: frontend/src/Models/Product.ts
import { SyntheticEvent } from "react";

export interface ProductGet {
  id: number;
  name: string;
  price: number;
  quatityInStock: number;
  description: string;
  image: string;
  supplierId: number; // Foreign key to Supplier
  categoryId: number; // Foreign key to category
}

export interface ProductPost {
  name: string;
  price: number;
  quatityInStock: number;
  description: string;
  image: string;
  supplierId: number; // Foreign key to Supplier
  categoryId: number; // Foreign key to category
}

File: api/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace api.Models
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        [Column(TypeName = "decimal(18,2)")]
        public decimal Price { get; set; }
        public int QuatityInStock { get; set; }
        public string Description { get; set; } = string.Empty;
        public string Image { get; set; } = string.Empty;
        public int? CategoryId { get; set; }
        public Category? Category { get; set; }
        public int? SupplierId { get; set; }
        public Supplier? Supplier { get; set; }
        public List<CartItem> CartItems { get; set; } = new List<CartItem>();

    }
}